engine/api/src/main/scala/pl/touk/nussknacker/engine/api/deployment/ProcessState.scala

```diff
@@ -1,21 +1,64 @@
 package pl.touk.nussknacker.engine.api.deployment
 
-import io.circe.{Decoder, Encoder}
 import io.circe.generic.JsonCodec
+import io.circe.{Decoder, Encoder, Json}
 import pl.touk.nussknacker.engine.api.ProcessVersion
+import pl.touk.nussknacker.engine.api.deployment.StateAction.StateAction
+import pl.touk.nussknacker.engine.api.deployment.StatusState.StateStatus
+import pl.touk.nussknacker.engine.defaults.deployment.DefaultProcessStateConfigurator
 
-@JsonCodec case class ProcessState(id: DeploymentId,
-                                   runningState: RunningState.Value,
+trait ProcessStateConfigurator {
```

Comment:
Please write comment here about usage - why maps and so on. WDYT about rename it to something with ProcessStateDefinitition in name? Like ProcessStateDefinitionProvider or ProcessStateDefinitionManager?

Reply:
Changed ProcessStateDefinitionManager.

Reply:
Added.